iwp_website/controllers/loan_issue.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2019-     Coop IT Easy SCRLfs <http://coopiteasy.be>
+#     - Rémy Taymans <remy@coopiteasy.be>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from werkzeug.exceptions import NotFound
+
+from odoo import http
+from odoo.http import request
+from odoo.tools.translate import _
+import logging
+
+
+_logger = logging.getLogger(__name__)
+
+
+class WebsiteLoanIssue(http.Controller):
+
+    @http.route(
+        [
+            '/loan/subscription',
+        ],
+        type='http',
+        auth='user',
+        website=True,
+    )
+    def subscribe_to_loan_issue(
+        self,
+        **post
+    ):
+        self.init_form_data(qcontext=post)
+        self.set_form_defaults(qcontext=post)
+        self.normalize_form_data(qcontext=post)
+        if post and request.httprequest.method == 'POST':
+            self.validate_form(qcontext=post)
+            if 'error' not in post:
+                values = self.prepare_loan_issue_line_value(qcontext=post)
+                request.env['loan.issue.line'].sudo().create(values)
+                post['success'] = True
+        # Populate template value
+        qcontext = post.copy()
+        return request.render(
+            'iwp_website.subscribe_to_loan_issue',
+            qcontext
+        )
+
+    def prepare_loan_issue_line_value(self, qcontext=None):
+        if qcontext is None:
+            qcontext = request.params
+        partner = request.env.user.partner_id
+        loan_issue = request.env['loan.issue'].browse(qcontext['loan_issue'])
+        values = {
+            'loan_issue_id': loan_issue.id,
+            'partner_id': partner.id,
+            'quantity': qcontext['quantity'],
+        }
+        return values
+
+    def validate_form(self, qcontext=None):
+        """
+        Populate request.parms with errors if the params from the form
+        are not correct. If a qcontext is given, this function works on
+        this qcontext.
+        """
+        if qcontext is None:
+            qcontext = request.params
+        selected_issue = qcontext.get('loan_issue', None)
+        if not selected_issue:
+            qcontext['error'] = _("You must select a financial product.")
+            return qcontext
+        if qcontext.get('quantity', 0) < 1:
```

Comment:
`elif`

Reply:
`if` because there is a return statement before. See no-else-return (R1705) in https://docs.pylint.org/en/1.9/technical_reference/features.html